src/tools/compiletest/src/runtest.rs

```diff
@@ -1896,6 +1896,84 @@ actual:\n\
         if !res.status.success() {
             self.fatal_proc_rec("htmldocck failed!", &res);
         }
+        self.check_rustdoc_test_option();
```

Comment:
This... is not what we want.

This means that **every single rustdoc test will run every `rustdoc-test` test**. Please integrate into the existing framework of compiletest in the same way that all other test are integrated. That means that test go in `src/test/rustdoc`, they've got a header indicating that what flavor of test it is, and then the test executes that.